### DISCUSSION TITLE
zh/guide/drivers/pikpak

### DISCUSSION THREAD

**Discussion by giscus** [AlistGo/docs#13] [Announcements]
# zh/guide/drivers/pikpak

AList v3的文档

https://alist.nn.ci/zh/guide/drivers/pikpak.html

**Comment by nifeng36** [comment]
请问invalid_request  怎么解决

**Reply by 132567** [reply]
解决了吗，哥们

**Reply by SHutdog** [reply]
同一样问题

**Comment by HedioKojima** [comment]
pikpak提供的这个网站无法使用了，还可以怎么获取根文件夹ID，谢谢

**Reply by tjsky** [reply]
随便挑一个就行
https://tjsky.github.io/pikpak/
https://fanghuan27.github.io/pikpak/

**Reply by loomee** [reply]
其实最新版alist如果没有特别需求，添加网盘的时候，跟文件夹id保持默认就好。
填写下挂载路径，这样就能把网盘主目录/根目录列出在所填写的文件夹内。
感觉官方教程好像特别强调要填写跟文件夹ID，个人觉得填写好挂载路径就很方便了。

**Reply by 49656374** [reply]
默认的root不行啊，只能定位到某一个文件夹了

**Reply by 220284** [reply]
我挂载完打开  只是定位到其中一个文件夹，请问如何解决呀

**Comment by bloding654** [comment]
太难了 在境外用国内的盘速度很慢 不知道是不是网络限制的原因还是其它什么原因。。想用谷歌和pikpak 根目录又获取不了 ~~

**Reply by Chengfanfu** [reply]
pikpak加载不出来，可能是网络的原因，我的电信网络正常访问，要是移动网络是无法访问的，很看地区网络，试试科学到新加坡访问

**Comment by choateyang** [comment]
3.2.1版本，挂载pikpak，302重定向。
下载文件，文件名不是原文件的，变成下载1、下载2这种

**Reply by heitu001** [reply]
我也是这样子

**Reply by unify-z** [reply]
这个是PikPak限制，不是alist的问题

**Comment by Mryiwen** [comment]
Post "https://user.mypikpak.com/v1/auth/signin": dial tcp: lookup user.mypikpak.com on 192.168.3.5:53: server misbehaving
请问群晖这个问题怎么解决？

**Comment by cosren** [comment]
添加失败 总是报错 ：Failed init storage: invalid_account_or_password
实际 账号密码 登录没问题

**Reply by zjokk** [reply]
我也是，不知朋友搞定了没有

**Comment by pipinian** [comment]
能登录查看文件，但复制时出现invalid_grant怎么解决

**Comment by megumi335400** [comment]
要是能支持facebook账密登录就好了，之前的账号就是用fb登录的，用fb的账密直接登不行

**Comment by zhongjunchen** [comment]
添加失败 总是报错 ：Failed init storage: invalid_account_or_password

对于这个问题，估计是手机登录出错，可以尝试在账号前加上国家编号，如+86，亲测有用，能正常添加以及访问。

**Reply by cosren** [reply]
多谢 ，问题解决 。就是要在 账号前面 加上 +86

**Comment by linzxc2010** [comment]
加载了但是播放视频很卡，需要什么特殊设置没有？

**Reply by zi4dbek** [reply]
别用移动宽带
或者使用魔法

**Comment by teddychgg** [comment]
pikpak挂在显示failed get objs: failed to list objs: invalid_argument；
怎么解决

**Reply by cosren** [reply]
就是要在 账号前面 加上 +86

**Reply by teddychgg** [reply]
在挂载的时候就加了+86，要不挂不上去，我现在是挂载上去了，但是打开文件夹报错，用邮箱登录一样

**Reply by real-jiakai** [reply]
AList管理—>存储—>重新加载全部，我的pikpak就挂载上去了。

**Comment by sparereal** [comment]
10T空间到期后，pikpak分享挂载后访问显示：failed get objs: failed to list objs: Sorry, sharing is not available in the current region ，求大神指点

**Reply by sparereal** [reply]
有时还显示：failed get link: share_status_prohibited

**Reply by duliri888** [reply]
解决了 吗  我和你一样报这个错

**Reply by ZJF-Kedaya** [reply]
我也这样解决了吗

**Comment by DowneyP** [comment]
你好 我想请问pikpak成功挂载后 能看到文件夹 也能看到文件 但是无法播放总是显示重新连接 1 ,2 ,3,4...等等 该怎么办呢?

**Reply by wsgehaishan520** [reply]
哈哈，解决了吗

**Comment by fengxiansen9** [comment]
求助梅林alist挂载pikpak出现Failed init storage: Post "https://user.mypikpak.com/v1/auth/signin": dial tcp ************3:443: connect: connection timed out。请问怎么解决

**Comment by zhuangyin8** [comment]
v3.14.0 无法从WebDAV下载文件

2023/03/24 20:51:08.510978 ERROR RESTY Get "https://api-drive.mypikpak.com/drive/v1/files?filters=%7B%22phase%22%3A%7B%22eq%22%3A%22PHASE_TYPE_COMPLETE%22%7D%2C%22trashed%22%3A%7B%22eq%22%3Afalse%7D%7D&limit=100&page_token=&parent_id=&thumbnail_size=SIZE_LARGE&with_audit=true": net/http: TLS handshake timeout, Attempt 1

**Reply by zhuangyin8** [reply]
INFO[2023-03-25 17:18:28] success load storage: [/PikPak], driver: [PikPak] 
2023/03/25 17:19:32.444579 ERROR RESTY context deadline exceeded (Client.Timeout or context cancellation while reading body), Attempt 1
2023/03/25 17:20:02.546787 ERROR RESTY context deadline exceeded (Client.Timeout or context cancellation while reading body), Attempt 2
2023/03/25 17:20:32.677769 ERROR RESTY Get "https://api-drive.mypikpak.com/drive/v1/files?filters=%7B%22phase%22%3A%7B%22eq%22%3A%22PHASE_TYPE_COMPLETE%22%7D%2C%22trashed%22%3A%7B%22eq%22%3Afalse%7D%7D&limit=100&page_token=eyJraW5kIjoidHlwZSIsImRhdGEiOnsidHlwZSI6MCwiZm9sZGVyX3R5cGUiOjAsImlkIjoiVk5SSkNaenpVVTluakpPUmh1cnUwc2FCbzEifX0&parent_id=&thumbnail_size=SIZE_LARGE&with_audit=true": context deadline exceeded (Client.Timeout exceeded while awaiting headers), Attempt 3
2023/03/25 17:20:33.372210 ERROR RESTY Get "https://api-drive.mypikpak.com/drive/v1/files?filters=%7B%22phase%22%3A%7B%22eq%22%3A%22PHASE_TYPE_COMPLETE%22%7D%2C%22trashed%22%3A%7B%22eq%22%3Afalse%7D%7D&limit=100&page_token=eyJraW5kIjoidHlwZSIsImRhdGEiOnsidHlwZSI6MCwiZm9sZGVyX3R5cGUiOjAsImlkIjoiVk5SSkNaenpVVTluakpPUmh1cnUwc2FCbzEifX0&parent_id=&thumbnail_size=SIZE_LARGE&with_audit=true": read tcp 10.66.203.106:49342->47.241.212.52:443: read: connection reset by peer, Attempt 4
2023/03/25 17:23:55.652849 ERROR RESTY Get "https://api-drive.mypikpak.com/drive/v1/files?filters=%7B%22phase%22%3A%7B%22eq%22%3A%22PHASE_TYPE_COMPLETE%22%7D%2C%22trashed%22%3A%7B%22eq%22%3Afalse%7D%7D&limit=100&page_token=eyJraW5kIjoidHlwZSIsImRhdGEiOnsidHlwZSI6MCwiZm9sZGVyX3R5cGUiOjAsImlkIjoiVk5IZkZvQmY3Wk1qZm9rd0dwdHFRa3Z4bzEifX0&parent_id=VNQz3jlffHAvkq8O_gVmpo3ao1&thumbnail_size=SIZE_LARGE&with_audit=true": context deadline exceeded (Client.Timeout exceeded while awaiting headers), Attempt 1
2023/03/25 17:23:55.691063 ERROR RESTY Get "https://api-drive.mypikpak.com/drive/v1/files?filters=%7B%22phase%22%3A%7B%22eq%22%3A%22PHASE_TYPE_COMPLETE%22%7D%2C%22trashed%22%3A%7B%22eq%22%3Afalse%7D%7D&limit=100&page_token=&parent_id=VND2QNF2hcn-RrsRA2UKY5zjo1&thumbnail_size=SIZE_LARGE&with_audit=true": context deadline exceeded (Client.Timeout exceeded while awaiting headers), Attempt 1
2023/03/25 17:24:25.753909 ERROR RESTY Get "https://api-drive.mypikpak.com/drive/v1/files?filters=%7B%22phase%22%3A%7B%22eq%22%3A%22PHASE_TYPE_COMPLETE%22%7D%2C%22trashed%22%3A%7B%22eq%22%3Afalse%7D%7D&limit=100&page_token=eyJraW5kIjoidHlwZSIsImRhdGEiOnsidHlwZSI6MCwiZm9sZGVyX3R5cGUiOjAsImlkIjoiVk5IZkZvQmY3Wk1qZm9rd0dwdHFRa3Z4bzEifX0&parent_id=VNQz3jlffHAvkq8O_gVmpo3ao1&thumbnail_size=SIZE_LARGE&with_audit=true": context deadline exceeded (Client.Timeout exceeded while awaiting headers), Attempt 2
2023/03/25 17:24:25.792248 ERROR RESTY Get "https://api-drive.mypikpak.com/drive/v1/files?filters=%7B%22phase%22%3A%7B%22eq%22%3A%22PHASE_TYPE_COMPLETE%22%7D%2C%22trashed%22%3A%7B%22eq%22%3Afalse%7D%7D&limit=100&page_token=&parent_id=VND2QNF2hcn-RrsRA2UKY5zjo1&thumbnail_size=SIZE_LARGE&with_audit=true": context deadline exceeded (Client.Timeout exceeded while awaiting headers), Attempt 2
2023/03/25 17:25:51.804884 ERROR RESTY Get "https://api-drive.mypikpak.com/drive/v1/files?filters=%7B%22phase%22%3A%7B%22eq%22%3A%22PHASE_TYPE_COMPLETE%22%7D%2C%22trashed%22%3A%7B%22eq%22%3Afalse%7D%7D&limit=100&page_token=eyJraW5kIjoidHlwZSIsImRhdGEiOnsidHlwZSI6MCwiZm9sZGVyX3R5cGUiOjAsImlkIjoiVk5QWm83UWtiR3BSQ2paMHdVTFRaVGhFbzEifX0&parent_id=VNPZo7QZbGpRCjZ0wULTZT23o1&thumbnail_size=SIZE_LARGE&with_audit=true": context deadline exceeded (Client.Timeout exceeded while awaiting headers), Attempt 1
2023/03/25 17:25:51.811174 ERROR RESTY Get "https://api-drive.mypikpak.com/drive/v1/files?filters=%7B%22phase%22%3A%7B%22eq%22%3A%22PHASE_TYPE_COMPLETE%22%7D%2C%22trashed%22%3A%7B%22eq%22%3Afalse%7D%7D&limit=100&page_token=&parent_id=VNFq3beI65qZogaWAy6_1j8Zo1&thumbnail_size=SIZE_LARGE&with_audit=true": context deadline exceeded (Client.Timeout exceeded while awaiting headers), Attempt 1
2023/03/25 17:26:28.873803 ERROR RESTY Get "https://api-drive.mypikpak.com/drive/v1/files?filters=%7B%22phase%22%3A%7B%22eq%22%3A%22PHASE_TYPE_COMPLETE%22%7D%2C%22trashed%22%3A%7B%22eq%22%3Afalse%7D%7D&limit=100&page_token=&parent_id=VNG9YH4zJkXSetoYWaixfOLPo1&thumbnail_size=SIZE_LARGE&with_audit=true": context deadline exceeded (Client.Timeout exceeded while awaiting headers), Attempt 1
2023/03/25 17:26:28.875998 ERROR RESTY Get "https://api-drive.mypikpak.com/drive/v1/files?filters=%7B%22phase%22%3A%7B%22eq%22%3A%22PHASE_TYPE_COMPLETE%22%7D%2C%22trashed%22%3A%7B%22eq%22%3Afalse%7D%7D&limit=100&page_token=eyJraW5kIjoidHlwZSIsImRhdGEiOnsidHlwZSI6MCwiZm9sZGVyX3R5cGUiOjAsImlkIjoiVk5QWm83UWhiR3BSQ2paMHdVTFRaVFlybzEifX0&parent_id=VNPZo7QZbGpRCjZ0wULTZT23o1&thumbnail_size=SIZE_LARGE&with_audit=true": context deadline exceeded (Client.Timeout exceeded while awaiting headers), Attempt 1
2023/03/25 17:26:34.774438 ERROR RESTY Get "https://api-drive.mypikpak.com/drive/v1/files/VNREe4fm5lP7OTIF6PnXIzNRo1?_magic=2021&thumbnail_size=SIZE_LARGE": context deadline exceeded (Client.Timeout exceeded while awaiting headers), Attempt 1
2023/03/25 17:43:21.816888 ERROR RESTY context deadline exceeded (Client.Timeout or context cancellation while reading body), Attempt 1
2023/03/25 17:43:22.205877 ERROR RESTY Get "https://api-drive.mypikpak.com/drive/v1/files/VNRK1AcsLmyyGDe21AoIsCTUo1?_magic=2021&thumbnail_size=SIZE_LARGE": read tcp 10.66.203.106:49736->47.241.212.52:443: read: connection reset by peer, Attempt 2
2023/03/25 17:48:32.164627 ERROR RESTY context deadline exceeded (Client.Timeout or context cancellation while reading body), Attempt 1
2023/03/25 17:48:32.564415 ERROR RESTY Get "https://api-drive.mypikpak.com/drive/v1/files/VNRGu4hc_oS16KEJoh-eZ2ieo1?_magic=2021&thumbnail_size=SIZE_LARGE": read tcp 10.66.203.106:49749->47.241.212.52:443: read: connection reset by peer, Attempt 2

**Comment by ddunniu** [comment]
Failed init storage: Post "https://user.mypikpak.com/v1/auth/signin": read tcp ***********82:57228->************2:443: wsarecv: An existing connection was forcibly closed by the remote host

**Comment by ddunniu** [comment]
failed get objs: failed to list objs: Get "https://api-drive.mypikpak.com/drive/v1/share/detail?filters=%7B%22phase%22%3A%7B%22eq%22%3A%22PHASE_TYPE_COMPLETE%22%7D%2C%22trashed%22%3A%7B%22eq%22%3Afalse%7D%7D&limit=100&page_token=&parent_id=&pass_code_token=&share_id=VNRUPOjF7aWNgw5Rd8ZqdENfo1&thumbnail_size=SIZE_LARGE&with_audit=true": read tcp ***********82:57899->************2:443: wsarecv: An existing connection was forcibly closed by the remote host.

**Reply by linger6580** [reply]
同样的问题，请问解决了吗

**Comment by Linjiangxian051** [comment]
请问挂载完成后，通过主页点进pikpak后，显示“failed get objs: failed to list objs: Sorry, sharing is not available in the current region”，怎么解决

**Comment by kevinzwq** [comment]
pikpak会员到期后挂载至网页的速度严重下降，各位大佬，有没有破解之法？

**Comment by TG-GOD** [comment]
failed get objs: failed to list objs: file_not_found,这个怎么解决

**Comment by JIUSUIAN** [comment]
请问这个怎么弄failed get objs: failed to list objs: file_not_found

**Comment by Archich** [comment]
請問分享自己的pikpak資料夾，如何查詢到自己的分享id
(為了不降速，我想把已填滿的10T帳號分享出來用6G帳號token觀賞)

**Comment by JIUSUIAN** [comment]
最近使用pikpak分享挂梯子速度很快，不挂梯子几乎没速度

**Comment by Terry-Wing** [comment]
问一下大家，开了pilpak会员了，挂载在Alist之后，播放pikpak里面的视频很慢啊，有什么解决方法吗？

**Comment by mapleshadow** [comment]
pikpak注定在国内不好使啊。。。。。。。。

**Comment by ZJF-Kedaya** [comment]
可以在alist内置一个代理，可以只代理pikpak，不影响其他盘的速度

**Reply by JIUSUIAN** [reply]
请教一下大佬，怎么内置呢

**Comment by freemer666** [comment]
挂载了PIKPAK后，不管是在Alist，还是PIKPAK软件里，一直提示无法播放视频什么原因呢？

**Comment by jameluo1996** [comment]
请问 No connection could be made because the target machine actively refused it怎么解决啊

**Comment by swenadolf** [comment]
Failed init storage: Post "https://user.mypikpak.com/v1/auth/signin": read tcp ************5:14205->***********4:443: wsarecv: An existing connection was forcibly closed by the remote host.


初始化存储失败，现有连接被远程主机强制关闭。这啥意思啊，

**Reply by linger6580** [reply]
同样问题

**Reply by swenadolf** [reply]
https://anwen-anyi.github.io/index/07-wenti.html#_40-%E4%B8%BA%E4%BB%80%E4%B9%88-terabox-%E6%8C%82%E8%BD%BD%E6%88%90%E5%8A%9F%E4%BA%86-%E4%BD%86%E6%98%AF%E4%B8%8D%E6%98%BE%E7%A4%BA%E5%86%85%E5%AE%B9
 常见问题：41

**Comment by at86397388** [comment]
用pikpak的app，速度就很快，但用alist挂载，速度就很慢，是什么原因呢

**Comment by zuilintan** [comment]
部分高清视频（非H265），
通过Pikpak挂载预览，会播放失败，
但通过SMB挂载预览，播放正常，
原因未知，但如此可以排除浏览器因素

**Reply by zuilintan** [reply]
更奇怪的是，在手机端夸克浏览器上打开链接，
播放正常

**Comment by kiolygenius** [comment]
2024/05/22 新出现 captcha_required 应该是需要登陆验证了。

**Reply by sirblangch** [reply]
那这种咋弄啊, 大哥你解决了吗

**Reply by 3182785169** [reply]
如何解决

**Reply by a5412398** [reply]
一样captcha_required	，目前好像还没有解决办法，等作者更新吧

**Comment by GaoHaiShan** [comment]
大佬 目前pikpak 在登陆前需要通过下面接口获取 captcha_token 了，能改一下逻辑吗
接口 ：https://user.mypikpak.com/v1/shield/captcha/init
参数
{
    "client_id": "YNxT9w7GMdWvEOKa",
    "action": "POST:/v1/auth/signin",
    "device_id": "a00737c5f9844055b9120d89789a8873",
    "meta": {
        "email": "xxxxxx"
    }
}

**Comment by zacblyhx** [comment]
登不上去了，需要验证码
Failed init storage: captcha_required 
求大佬改进

**Reply by AFeixue** [reply]
我也是这样

**Reply by sirblangch** [reply]
+1

**Comment by sirblangch** [comment]
请问这个怎么解决啊: Failed init storage: captcha_required

**Comment by lingchvip** [comment]
突然不能挂了，提示：Failed init storage: captcha_required，验证错误，咋弄了呢？

**Reply by 8mayongde2** [reply]
一样Failed init storage: captcha_required，期待解决方案。

**Comment by linep47** [comment]
如果是付费用户请打开webdav权限在网页版里打开，然后alist里面通过webdav挂载，链接选择https://dav.mypikpak.com/或http://dav.mypikpak.com/，免费用户没找到办法

**Reply by ZhengZhaoxi** [reply]
pikpak的webdav挂载是不是没法转存别的网盘的文件了

**Comment by lmx4727** [comment]
请问挂载pikpak出现这个怎么解决
Failed init storage but storage is already created: failed init storage: Post "https://user.mypikpak.com/v1/auth/token": dial tcp ***********83:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.

**Comment by chivehao** [comment]
不知道PKPK能否在alist端实现秒传磁力链接，
目前还需要先进复制番剧的磁链接，到PKPK的web端添加后，再去后台重启alist容器，才会有添加后的目录，
如果alist这边能直接添加秒传磁链就好了

**Comment by Donck0606** [comment]
客户端ID*
客户端密钥*
今天早上发现alist挂载pikpak出错了，发现多了“客户端ID*”和“客户端密钥*”两个选项，不知道怎么搞了。

**Reply by Gehinax** [reply]
Pikpak挂载访问不了有4K，2K清晰度的视频，也下载不了，只能访问和下载清晰度是原画的视频

**Comment by Gehinax** [comment]
Pikpak挂载访问不了有4K，2K清晰度的视频，也下载不了，只能访问和下载清晰度是原画的视频

**Comment by 8mayongde2** [comment]
昨天开始看不了pikpak内部资源，望早些解决。

**Comment by xiaobin1123** [comment]
当前版本v3.35.0，能够看到pikpak目录下的文件，但是无法复制到本地存储。没有任何报错，进度条一直是0.

**Reply by bluefantasy0830** [reply]
现在用魔法才能看了,不知道能不能解决

**Comment by guoyan1983** [comment]
Failed init storage: captcha_required，显示这个怎么回事啊

**Reply by stifredm** [reply]
更新一下alist本身就好了

**Comment by zwk104** [comment]
Failed init storage: captcha_required这个问题怎么解决

**Reply by stifredm** [reply]
更新一下alist本身就好了

**Comment by ycy655188** [comment]
挂在pikpak出现这样failed get objs: failed to list objs: oauth2: "invalid_argument" "currently not supported"该怎么办

**Comment by qdws** [comment]
failed get objs: failed to list objs: oauth2: "invalid_argument" "currently not supported"这个问题怎么解决？

**Comment by yongzheng909** [comment]
挂载pikpak时新出现了“客户端ID”和“客户端密钥”这些怎么填？

**Comment by qdws** [comment]
没有找到可以匹配的答案。

yongzheng909 ***@***.***> 于2024年8月17日周六 21:37写道：

**Comment by BurningSky0306** [comment]
最近几天开始，3.35.0版本和最新3.36.0版本挂载PikPak网盘时都会出现  failed get objs: failed to list objs: oauth2: "invalid_argument" "currently not supported"
  的报错，辛苦更新解决一下这个BUG🤓🤓🤓

**Comment by laucherish** [comment]
failed get objs: failed to list objs: oauth2: "invalid_argument" "currently not supported"
能否解决下这个问题

**Reply by iislong** [reply]
换beta版

**Comment by BurningSky0306** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我最近正在休假中，无法亲自回复你的邮件。我将在假期结束后，尽快给你回复。

**Comment by mu1989012303** [comment]
求助，添加时提示需要验证码，怎么办？

**Comment by landmarkgzz** [comment]
这问题怎么解决啊：初始化存储失败，但已创建存储：初始化存储失败：发布 “https://user.mypikpak.com/v1/auth/signin”：读取 tcp ***********29：7702->***********83：443：wsarecv：远程主机强制关闭了现有连接。

**Comment by 4o3-o** [comment]
pikpak限制规则之后，直接用webdav方式挂载pikpak就可以了，稳定高效

**Reply by RotteSya** [reply]
记得官方提供的webdav服务不是不允许写入吗

**Reply by 4o3-o** [reply]
目前确实不允许写入，但挂载alist的文件更多场景是读取，至少解决了pikpak账号挂载无法顺利读取和下载的问题，而且更新规则以后在大陆使用必须用代理服务器下载文件，有点吃不消，挂载到境外服务器的alist上下载是不需要代理的

**Comment by Mryiwen** [comment]
群晖的cloud sync用不了官方webdav无法备份至群辉差了意思RotteSya ***@***.***>於2024年9月12日 23:09寫道：﻿
记得官方提供的webdav服务不是不允许写入吗

**Reply by 4o3-o** [reply]
pikpak挂alist，群晖可以连alist的webdav，就可以备份了

**Comment by Mryiwen** [comment]
现在要怎么配置了？上个版本失效到现在都没设置好Orange ***@***.***>於2024年9月13日 13:39寫道：﻿
pikpak挂alist，群晖可以连alist的webdav，就可以备份了

**Reply by 4o3-o** [reply]
首先你要有pikpak的会员，在pikpak设置中有个实验室功能，里面有webdav相关设置和连接用的账号密码信息，启用这个功能；alist用webdav的方式挂载pikpak，比如你的pikpak文件夹结构是在根目录有个“test”文件夹，那么挂载的时候就填路径“/test”即可，填写pikpak提供的webdav的连接地址、账号和密码就可以了；
alist的webdav功能有官方文档的，地址就是“你的alist域名/dav”，如果你要挂载特定的目录，那可以写成“你的alist域名/dav/test”，比如这里就是把pikpak的test文件夹挂载到alist自定义的路径test中就这么写，账号密码就是你的alist管理员账号密码，直接连接就可以实现群晖备份pikpak文件的功能了，这里只是用了alist作为中间件处理群晖和pikpak的业务

**Comment by Mryiwen** [comment]
能来个配置图吗？pikpak不是会员就是废的Orange ***@***.***>於2024年9月13日 15:22寫道：﻿
首先你要有pikpak的会员，在pikpak设置中有个实验室功能，里面有webdav相关设置和连接用的账号密码信息，启用这个功能；alist用webdav的方式挂载pikpak，比如你的pikpak文件夹结构是在根目录有个“test”文件夹，那么挂载的时候就填路径“/test”即可，填写pikpak提供的webdav的连接地址、账号和密码就可以了；
alist的webdav功能有官方文档的，地址就是“你的alist域名/dav”，如果你要挂载特定的目录，那可以写成“你的alist域名/dav/test”，比如这里就是把pikpak的test文件夹挂载到alist自定义的路径test中就这么写，账号密码就是你的alist管理员账号密码，直接连接就可以实现群晖备份pikpak文件的功能了，这里只是用了alist作为中间件处理群晖和pikpak的业务

**Reply by 4o3-o** [reply]
这是详细设置方式，参考图内标注的信息，用的愉快。
alist设置
![b0016ad2bdd48cc559ac1895b22984f2.png](https://ice.frostsky.com/2024/09/13/b0016ad2bdd48cc559ac1895b22984f2.png)
群晖设置
![d3e58983572c5eddbfd1f50ea8c27780.png](https://ice.frostsky.com/2024/09/13/d3e58983572c5eddbfd1f50ea8c27780.png)
![65b33d6ade31bb69790e200d3465b2ce.png](https://ice.frostsky.com/2024/09/13/65b33d6ade31bb69790e200d3465b2ce.png)
![da5908a4a0d40f5a31513c46b625d72c.png](https://ice.frostsky.com/2024/09/13/da5908a4a0d40f5a31513c46b625d72c.png)

**Reply by Jjjeerry** [reply]
请问一下根文件夹路径填写什么，这里一直不对

**Reply by Lingoluchaoli** [reply]
哥，飞牛os怎么挂远程存储

**Reply by Lingoluchaoli** [reply]
Alist里面添加了webdav是work状态，但是点开又是错误端口443错误

**Reply by 4o3-o** [reply]
1. 根文件夹路径就是”/“，比如你挂载pikpak默认的文件夹”my pack“这个文件夹，就写入”/my pack“就可以了

2.飞牛os我没用过，演示的截图是synology的DSM系统

3.如果你的alist是安装在本地，且本地网络又恰好处在pikpak不支持的国家和地区，那么你应该让alist走代理网络挂载pikpak的webdav

**Comment by Mryiwen** [comment]
谢谢🙏Orange ***@***.***>於2024年9月13日 23:39寫道：﻿
这是详细设置方式，参考图内标注的信息，用的愉快。
alist设置

群晖设置

**Comment by workerspages** [comment]
Request failed with status code 500

**Reply by 4o3-o** [reply]
这应该是pikpak服务器问题，pikpak的服务在新加坡阿里云机房火灾期间受很大影响一度不可用，不过最新公告称恢复了，可以试试看

**Comment by Mryiwen** [comment]
﻿根目录填什么？是/root吗？Orange ***@***.***>於2024年9月13日 23:39寫道：﻿
这是详细设置方式，参考图内标注的信息，用的愉快。
alist设置

群晖设置

**Comment by qingyi0530** [comment]
Failed init storage: ErrorCode: 3 ,Error: invalid_argument ,ErrorDescription: x-client-id required

**Reply by 4o3-o** [reply]
这应该是pikpak服务器问题，pikpak的服务在新加坡阿里云机房火灾期间受很大影响一度不可用，不过最新公告称恢复了，可以试试看

**Comment by Mryiwen** [comment]
谢谢，确实很没挂科学网络的问题Orange ***@***.***>於2024年9月25日 00:23寫道：﻿

根文件夹路径就是”/“，比如你挂载pikpak默认的文件夹”my pack“这个文件夹，就写入”/my pack“就可以了

2.飞牛os我没用过，演示的截图是synology的DSM系统
3.如果你的alist是安装在本地，且本地网络又恰好处在pikpak不支持的国家和地区，那么你应该让alist走代理网络挂载pikpak的webdav

**Comment by lsCoding666** [comment]
现在我用这个存储类型感觉很奇怪，首先，如果你只需要读取的话，并且你有pikpak会员的话，确实使用pikpak自带的webdav挂载到alist上，就能用了。但是如果你需要写文件，那就不行了，因为官方的webdav是只允许读取的，不支持写入。

所以要写入的话，只能用pikpak账号密码、refresh token的方式，把pikpak挂载到alist上。但是这么搞的话，经常会出现refresh token过期的问题，部署上第一天能用，第二天就用不了，所以感觉如果是要写入的话，这个存储基本上就不可用

我自己的使用场景，是pikpak 挂载到alist上，然后再用rclone挂载alist的webdav到本地使用，然后脚本定时复制文件到挂载目录，用来备份文件使用。但是由于经常出现refresh token过期、被其他设备刷新的问题，导致这个方案基本上不可用，所以我想了解下有没有改进的方法，因为写不了确实很蛋疼。

**Reply by lsCoding666** [reply]
后续：改用阿里云对象存储了，免费5g，备份的话也无流量费用

**Reply by lsCoding666** [reply]
写个脚本只保留最新的2份

**Comment by vpei** [comment]
Failed load storage: failed init storage: Post "https://user.mypikpak.com/v1/auth/token?client_id=YUMx5nI8ZU8Ap8pn": tls: failed to verify certificate: x509: certificate is not valid for any names, but wanted to match user.mypikpak.com

怎么出错了？

**Reply by vpei** [reply]
failed get objs: failed to list objs: Get "https://api-drive.mypikpak.com/drive/v1/files?filters=%7B%22phase%22%3A%7B%22eq%22%3A%22PHASE_TYPE_COMPLETE%22%7D%2C%22trashed%22%3A%7B%22eq%22%3Afalse%7D%7D&limit=100&page_token=&parent_id=VNyIu3SCXFvgpfo5PjfvcZE5o1&thumbnail_size=SIZE_LARGE&with_audit=true": tls: failed to verify certificate: x509: certificate is not valid for any names, but wanted to match api-drive.mypikpak.com
